### DISCUSSION TITLE
How to use Paging extension?

### DISCUSSION THREAD

**Discussion by msfjarvis** [dropbox/Store#642] [Q&A] [answered]
I have a paginated API that I was hoping to use Store with, but the current `StoreKey` / `StoreData` have me quite perplexed (I've already seen https://github.com/MobileNativeFoundation/Store/issues/604).

PR #550 that introduced this API rather unfortunately chose to completely omit tests/samples for the `Page` based API.

I can not navigate this maze of generics for the life of me. `StoreKey` requires a generic `Id`, but if I choose to use `StoreKey.Collection.Page` it just automatically gets hard-coded to `Nothing`. When I try to implement `StoreData.Collection`, I am forced to also provide a `StoreData.Single<Id>` which my API will never return. After I end up exposing a field from my data model as the necessary `Id` type, I find that the `launchPagingStore` extension does not compile because some generic type bound is unsatisfied. 

Here are all the changes I've made so far, the relevant bits are in the `store` module: https://github.com/msfjarvis/compose-lobsters/compare/main...hs/newest-posts-store

I'd like either some help with making this work, or an answer for whether I should simply wait for the newer paging API and shelve all this work.

**Comment by matt-ramotar** [comment, marked as answer]
Hey there. Yes, please hold off until the newer paging API. I gave a talk on it at Droidcon SF last week. Aiming to release it in the next week or so

**Reply by msfjarvis** [reply]
Is there anything I can help with to move this along?

**Reply by matt-ramotar** [reply]
Sorry to be slow, no I don't think so unfortunately. Very busy few months, trying to get it in a state for review
